apps/pipelines/nodes/nodes.py

```diff
@@ -447,35 +456,49 @@ class RouterNode(RouterMixin, Passthrough, HistoryMixin):
             field_order=["llm_provider_id", "llm_temperature", "history_type", "prompt", "keywords"],
         )
     )
-
     prompt: str = Field(
         default="You are an extremely helpful router",
         min_length=1,
         json_schema_extra=UiSchema(widget=Widgets.expandable_text),
     )
 
     def _process_conditional(self, state: PipelineState, node_id=None):
-        from apps.service_providers.llm_service.prompt_context import PromptTemplateContext
-
-        prompt = OcsPromptTemplate.from_messages(
+        prompt = ChatPromptTemplate.from_messages(
             [("system", self.prompt), MessagesPlaceholder("history", optional=True), ("human", "{input}")]
         )
 
         session: ExperimentSession = state["experiment_session"]
         node_input = state["messages"][-1]
+
         context = {"input": node_input}
         context.update(PromptTemplateContext(session).get_context(prompt.input_variables))
 
         if self.history_type != PipelineChatHistoryTypes.NONE and session:
             input_messages = prompt.format_messages(**context)
             context["history"] = self._get_history(session, node_id, input_messages)
 
-        chain = prompt | self.get_chat_model()
-
+        llm = self.get_chat_model()
+        router_schema = self._create_router_schema()
+        chain = prompt | llm.with_structured_output(router_schema)
         result = chain.invoke(context, config=self._config)
-        keyword = self._get_keyword(result.content)
+
+        valid_keywords = [k.lower() for k in self.keywords]
+        default_keyword = self.keywords[0].lower()
+        try:
+            keyword = (
+                getattr(result, "route", None)
+                or (result.get("route") if isinstance(result, dict) else None)
+                or self._get_keyword(str(result))
```

Comment:
Should `result.get("route")` not always work?

Reply:
it failed in tests without adding the rest which is why I added, but open to suggestions

Reply:
great point: [fd566ab](https://github.com/dimagi/open-chat-studio/pull/1403/commits/fd566abf41dcc75e8f8bfd5f8c93165d63d048e3)